### ISSUE TITLE
Python environments are activated via dot source

### ISSUE THREAD

**Issue by DownTownScarf7** [returntocorp/bento#232]
## Overview
> Briefly describe the issue and your expected behavior

Environments are activated via dot sourcing (`. venv/bin/activate`) while they should be sourced (`source venv/bin/activate`)

## Current Behavior
> Describe the actual behavior, and provide the content of `~/.bento/last.log` if possible. Screenshots and text dumps are particularly helpful.

dot sourcing creates a subshell, which do not have environmental variables set such as proxies, getting connection error due to proxies not being set

```log
[DEBUG] 369443.09878349304 root:python_tool - r2c.bandit: Running '. 
'/home/bence/Projects/Ymir/.bento/bandit/bin/activate'; ...
[ERROR] 556.4303398132324 root:network - HTTPSConnectionPool(host='bento.r2c.dev', port=443): Max retries exceeded with url: /bento/api/v1/version (Caused by ProxyError('Cannot connect to proxy.', timeout('timed out')))
```

## Steps to Reproduce

1. bento check
2. ???
3. profit

## Possible Solution
> Do you have an idea for how to address the issue or feedback?

1. use `source`
2. dump environmental variables into a file, prepend them with export, append file with the dot sourcing command
3. `http_proxy=http_proxy https_proxy=https_proxy ftp_proxy=ftp_proxy bash -c '. /path/to/venv/bin/activate'`
4. prepend all `activate` files with exporting currently set proxies

If there is no major reason why you are not doing `1.`, please do not even consider `2.`, `3.` or `4.`

## Workaround

My current workaround is `4.` from "Possible Solution"

## Additional Context
Bento version (run `bento --version`): bento/0.7.0
Operation system (e.g. Ubuntu 18.04): Debian GNU/Linux 10 (buster)
Any other context that would help us reproduce or understand the issue: 

[source man page](https://ss64.com/bash/source.html)

(very basic TL;DR): `./file` executes file in subshell if execute permission is set, `. ./file` executes file in subshell regardless of execute permission, `source ./file` executes file in current shell

**Comment by deifactor** [comment]
I think we can just manually set the environment variables the venv activates ourselves, since the activate script only sets:

- `VIRTUAL_ENV` (not sure if this is used, but can't hurt)
- `PATH` (definitely want this one)
- unsets `PYTHONHOME` (definitely want this one)
- `PS1` (not important)

**Comment by deifactor** [comment]
(Also, thank you for such in-depth investigation and feedback!)

**Comment by DownTownScarf7** [comment]
> I think we can just manually set the environment variables the venv activates ourselves, since the activate script only sets:
> 
> * `VIRTUAL_ENV` (not sure if this is used, but can't hurt)
> * `PATH` (definitely want this one)
> * unsets `PYTHONHOME` (definitely want this one)
> * `PS1` (not important)

Hmn, I think you are thinking about this backwards or I am missing something.
Problem is not that the shell doesn't have the subshell environmental variables, it's that the subshell doesn't have the shell's environmental variables.

My concrete problem right now is that `http_proxy`, `https_proxy`, `ftp_proxy` and `no_proxy` are set in the shell but, however dot sourcing creates a subshell which do not carry environmental variables over, so I have no proxies set there, which makes Bento give me `ProxyError`s.

Now, even in this situation you could set the variables manually, one-by-one, but I do believe that this issue will show up from time to time, because there could be a few interesting things set as environmental vars. The first line of the `activate` file is not a recommendation but an instruction.

**Comment by deifactor** [comment]
Yeah, let me clarify. The problem doesn't have anything to do with `source` vs `.`, since those are actually identical. The problem is that when we were executing the bandit/flake8 subcommands, we were running them in a *blank* environment (modulo some other things that aren't relevant to this bug).

So the fix is to inherit the parent environment by default, by copying `os.environ`, making the changes we needed, and then passing that as the `env` argument to `subprocess.run`. But while I was making that fix, I realized that this was the perfect opportunity to make another fix I'd had in mind for a while: since all the `activate` script really does is modify env variables, we can just make those changes to the `env` argument ourselves.

So the code now looks like

```
env = dict(os.environ)
env["VIRTUAL_ENV"] = ...
env["PATH"] = ...
subprocess.run(
  cmd=...,
  env=env,
  ...
)
```

I've tested this, it works, it just hasn't made it into a release yet.
